### PR TITLE
AP_Windvane: add a rpm sensor type and AP_RPM: create singleton 

### PR DESCRIPTION
**Pull request by IamPete1**
This adds a singleton to AP_RPM and uses it in AP_Windvane for a 'cups' type spinning anemometer. RPM_SCALING parameter must be set such that the rpm sensor returns the wind speed in m/s. AP_Windvane uses the first RPM sensor. 

Not been tested but its not a drastic change.

### CODE REVIEW

File: libraries/AP_WindVane/AP_WindVane.cpp [hunk 1]
```diff
@@ -374,6 +374,16 @@ void AP_WindVane::update_apparent_wind_speed()
             apparent_speed_in = read_wind_speed_SITL();
 #endif
             break;
+        case WINDSPEED_RPM: {
+            AP_RPM* rpm = AP_RPM::get_singleton();
```
Comment (peterbarker) [inline libraries/AP_WindVane/AP_WindVane.cpp]: `const`

File: libraries/AP_WindVane/AP_WindVane.cpp [hunk 2]
```diff
@@ -118,7 +118,7 @@ const AP_Param::GroupInfo AP_WindVane::var_info[] = {
     // @Param: SPEED_TYPE
     // @DisplayName: Wind speed sensor Type
     // @Description: Wind speed sensor type
-    // @Values: 0:None,1:Airspeed library,2:Moden Devices Wind Sensor,3:SITL
```
Comment (peterbarker) [inline libraries/AP_WindVane/AP_WindVane.cpp]: What happened to the SITL backend?

Comment (IamPete1) [reply libraries/AP_WindVane/AP_WindVane.cpp]: the parameters had it at 3 but it was 10 in the enum so i have updated the description to match

File: libraries/AP_WindVane/AP_WindVane.cpp (hunk 1 above)
Comment (IamPete1) [reply libraries/AP_WindVane/AP_WindVane.cpp]: fixed

**Comment by IamPete1** [comment]
looks like I got in-between something broken on master when I fixed, semaphoreci failing is unrelated to this

**Comment by WickedShell** [comment]
@IamPete1 if you rebase (assuming Master is good) the issue should disappear.

On a technical level, Travis tests a rebased version of the PR, while semaphore tests the PR exactly as it's sent in.

**Comment by IamPete1** [comment]
yep, passed this time, thanks

File: libraries/AP_WindVane/AP_WindVane.cpp (hunk 2 above)
Comment (rmackay9) [reply libraries/AP_WindVane/AP_WindVane.cpp]: I think we should probably shorten the string for "3" to just "RPM sensor" and then add extra info to the wiki.

**Comment by rmackay9** [comment]
Looks good to me except for the slight long string which I've comment on.  Fix that and I'll merge, thanks!

File: libraries/AP_WindVane/AP_WindVane.cpp (hunk 2 above)
Comment (IamPete1) [reply libraries/AP_WindVane/AP_WindVane.cpp]: fixed,  i will do a PR to update the wiki also

File: libraries/AP_WindVane/AP_WindVane.cpp [hunk 3]
```diff
@@ -289,7 +289,7 @@ float AP_WindVane::read_SITL_direction_ef()
     const float wind_speed = AP::sitl()->wind_speed_active;
     const float wind_dir_rad = radians(AP::sitl()->wind_direction_active);
 
-    // Note than the SITL wind direction is defined as the direction the wind is travelling to
+    // Note than the SITL wind direction is defined as the direction the wind is traveling to
```
Comment (rmackay9) [inline libraries/AP_WindVane/AP_WindVane.cpp]: apparently the difference between "traveling" and "travelling" is just a UK vs US thing..  it's fine to change it though.

Comment (IamPete1) [reply libraries/AP_WindVane/AP_WindVane.cpp]: I worked out how to do spellcheck in notepad++ after doing the original sailboat stuff, after you had to correct so much of my bad spelling. This was picked up by the spellchecker, not sure what version of English I have it set to tho

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**